fv3gfs/physics/stencils/microphysics.py

```diff
@@ -1756,165 +1797,67 @@ def fields_update(
 
 
 class MicrophysicsState:
+    """
+    pt, qvapor, qrain, qice, qsnow, qgraupel, qcld, &
+    ua, va, delp, delz, omga: same as physics state
+    qv_dt: specific humidity tendency
+    ql_dt: cloud water mixing ratio tendency
+    qr_dt: rain water mixing ratio tendency
+    qi_dt: ice water mixing ratio tendency
+    qs_dt: snow mixing ratio tendency
+    qg_dt: graupel mixing ratio tendency
+    qa_dt: cloud fraction tendency
+    udt: eastard wind tendency
+    vdt: northward wind tendency
+    pt_dt: air temperature tendency
+    """
+
     def __init__(
         self,
-        grid,
-        air_temperature: Quantity,
-        specific_humidity: Quantity,
-        cloud_water_mixing_ratio: Quantity,
-        rain_mixing_ratio: Quantity,
-        cloud_ice_mixing_ratio: Quantity,
-        snow_mixing_ratio: Quantity,
-        graupel_mixing_ratio: Quantity,
-        cloud_fraction: Quantity,
-        eastward_wind: Quantity,
-        northward_wind: Quantity,
-        pressure_thickness_of_atmospheric_layer: Quantity,
-        vertical_thickness_of_atmospheric_layer: Quantity,
-        vertical_pressure_velocity: Quantity,
+        pt: FloatField,
+        qvapor: FloatField,
+        qliquid: FloatField,
+        qrain: FloatField,
+        qice: FloatField,
+        qsnow: FloatField,
+        qgraupel: FloatField,
+        qcld: FloatField,
+        ua: FloatField,
+        va: FloatField,
+        delp: FloatField,
+        delz: FloatField,
+        omga: FloatField,
+        delprsi: FloatField,
+        wmp: FloatField,
+        dz: FloatField,
+        storage: FloatField,
```

Comment:
Can we figure a better variable name than `storage`? Maybe something that describes the layout? Or something like `species_storage`?

Reply:
Alternatively you could pass a `shape`/`dtype`

Reply:
changed to `tendency_storage`